src/components/ArtifactProductMilestonesReleasesList/ArtifactProductMilestonesReleasesList.tsx

```diff
@@ -0,0 +1,134 @@
+import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { Link } from 'react-router-dom';
+
+import { MilestoneInfo } from 'pnc-api-types-ts';
+
+import { artfactProductMilestoneReleaseEntityAttributes } from 'common/artfactProductMilestoneReleaseEntityAttributes';
+import { PageTitles } from 'common/constants';
+import { getFilterAttributes } from 'common/entityAttributes';
+
+import { IServiceContainer } from 'hooks/useServiceContainer';
+import { ISortAttributes, useSorting } from 'hooks/useSorting';
+
+import { ContentBox } from 'components/ContentBox/ContentBox';
+import { Filtering } from 'components/Filtering/Filtering';
+import { Pagination } from 'components/Pagination/Pagination';
+import { ServiceContainerLoading } from 'components/ServiceContainers/ServiceContainerLoading';
+import { Toolbar } from 'components/Toolbar/Toolbar';
+import { ToolbarItem } from 'components/Toolbar/ToolbarItem';
+
+import { createDateTime } from 'utils/utils';
+
+const sortAttributes: ISortAttributes = {
+  milestoneEndDate: {
+    id: 'milestoneEndDate',
+    title: 'Completed On',
+    tableColumnIndex: 3,
+  },
+  releaseReleaseDate: {
+    id: 'releaseReleaseDate',
+    title: 'Released On',
+    tableColumnIndex: 5,
+  },
+};
+
+interface IArtifactProductMilestonesReleasesListProps {
+  serviceContainerArtifactProductMilestonesReleases: IServiceContainer;
+  componentId: string;
+}
+
+/**
+ * Component displaying list of Product Milestones that produced or consumed an Artifact.
+ *
+ * @param serviceContainerArtifactProductMilestonesReleases - Service Container for Product Milestone info
+ * @param componentId - Component ID
+ */
+export const ArtifactProductMilestonesReleasesList = ({
+  serviceContainerArtifactProductMilestonesReleases,
+  componentId,
+}: IArtifactProductMilestonesReleasesListProps) => {
+  const { getSortParams } = useSorting(sortAttributes, componentId);
+
+  return (
+    <>
+      <Toolbar>
+        <ToolbarItem>
+          <Filtering
+            filterOptions={getFilterAttributes(artfactProductMilestoneReleaseEntityAttributes)}
+            componentId={componentId}
+          />
+        </ToolbarItem>
+      </Toolbar>
+
+      <ContentBox borderTop>
+        <ServiceContainerLoading {...serviceContainerArtifactProductMilestonesReleases} title={PageTitles.productMilestones}>
+          <TableComposable isStriped variant="compact">
+            <Thead>
+              <Tr>
+                <Th width={20}>Product {artfactProductMilestoneReleaseEntityAttributes.productName.title}</Th>
```

Comment:
In the end `Product` prefix needs to be moved into `artfactProductMilestoneReleaseEntityAttributes` to make sure both filtering and column header titles are matching.

Reply:
done